tests/PHPStan/Analyser/data/integer-range-types.php

```diff
@@ -218,70 +218,70 @@ public function math($i, $j, $z, $pi, $r1, $r2, $r3, $rMin, $rMax, $x, $y) {
 		assertType('int<2, 13>', $r1 + $j);
 		assertType('int<-2, 9>', $r1 - $j);
 		assertType('int<1, 30>', $r1 * $j);
-		assertType('float|int<0, 10>', $r1 / $j);
+		assertType('float|int<1, 10>', $r1 / $j);
 		assertType('int<min, 15>', $rMin * $j);
 		assertType('int<5, max>', $rMax * $j);
 
 		assertType('int<2, 13>', $j + $r1);
 		assertType('int<-9, 2>', $j - $r1);
 		assertType('int<1, 30>', $j * $r1);
-		assertType('float|int<0, 3>', $j / $r1);
+		assertType('float|int<1, 3>', $j / $r1);
 		assertType('int<min, 15>', $j * $rMin);
 		assertType('int<5, max>', $j * $rMax);
 
 		assertType('int<-19, -10>|int<2, 13>', $r1 + $z);
 		assertType('int<-2, 9>|int<21, 30>', $r1 - $z);
 		assertType('int<-200, -20>|int<1, 30>', $r1 * $z);
-		assertType('float|int<0, 10>', $r1 / $z);
+		assertType('float|int<1, 10>', $r1 / $z);
 		assertType('int', $rMin * $z);
 		assertType('int<min, -100>|int<5, max>', $rMax * $z);
 
 		assertType('int<2, max>', $pi + 1);
 		assertType('int<-1, max>', $pi - 2);
 		assertType('int<2, max>', $pi * 2);
-		assertType('float|int<0, max>', $pi / 2);
+		assertType('float|int<1, max>', $pi / 2);
 		assertType('int<2, max>', 1 + $pi);
 		assertType('int<min, 2>', 2 - $pi);
 		assertType('int<2, max>', 2 * $pi);
-		assertType('float|int<2, max>', 2 / $pi);
+		assertType('float|int<1, 2>', 2 / $pi);
 
 		assertType('int<5, 14>', $r1 + 4);
 		assertType('int<-3, 6>', $r1 - 4);
 		assertType('int<4, 40>', $r1 * 4);
-		assertType('float|int<0, 2>', $r1 / 4);
+		assertType('float|int<1, 2>', $r1 / 4);
 		assertType('int<9, max>', $rMax + 4);
 		assertType('int<1, max>', $rMax - 4);
 		assertType('int<20, max>', $rMax * 4);
-		assertType('float|int<1, max>', $rMax / 4);
+		assertType('float|int<2, max>', $rMax / 4);
 
 		assertType('int<6, 20>', $r1 + $r2);
 		assertType('int<-9, 5>', $r1 - $r2);
 		assertType('int<5, 100>', $r1 * $r2);
-		assertType('float|int<0, 1>', $r1 / $r2);
+		assertType('float|int<1, 2>', $r1 / $r2);
 
 		assertType('int<-99, 19>', $r1 - $r3);
 
 		assertType('int<min, 15>', $r1 + $rMin);
 		assertType('int<-4, max>', $r1 - $rMin);
 		assertType('int<min, 50>', $r1 * $rMin);
-		assertType('float|int<min, 2>', $r1 / $rMin);
+		assertType('float|int<-10, -1>|int<1, 10>', $r1 / $rMin);
```

Comment:
int<1, 10> / int<min, 5>

10 / -1 = -10 and 10 / -10 = -1
and 10 / 10 = 1 and 10 / 1 = 10

Reply:
Woflramalpha is not doing great on this https://www.wolframalpha.com/input?i=%28interval%5B1%2C10%5D+%2F+interval%5B-9999%2C5%5D%29